core/trino-main/src/main/java/io/trino/FeaturesConfig.java

```diff
@@ -491,4 +496,17 @@ public FeaturesConfig setForceSpillingJoin(boolean forceSpillingJoin)
         this.forceSpillingJoin = forceSpillingJoin;
         return this;
     }
+
+    public int getJoinPositionsArrayCutoff()
+    {
+        return joinPositionsArrayCutoff;
+    }
+
+    @Config("join-positions-array-cutoff")
```

Comment:
Property name is very vague. I would rather have a kill switch as for ```incremental-hash-array-load-factor.enabled```

Reply:
I disagree. With a configurable value the user can sacrifice some memory for a faster join. I believe many deployments have some spare memory and can use a speedup.
We can use it in benchmarking as well.

Reply:
> I disagree. With a configurable value the user can sacrifice some memory for a faster join. I believe many deployments have some spare memory and can use a speedup.
We can use it in benchmarking as well.

I don't think anyone will ever change it. Let's just target specific use case (small build side) which seems like a majority of joins

Reply:
We can benchmark different values in the future and see how it affects performance. The tpch q09 can get a 10% speed up when this value is increased. IMHO There are many less important config options that remain in Trino and there is no harm in having them.
> (small build side) which seems like a majority of joins

This is not really true. The joins with big build side are maybe not so common, but they are wasting the most of cpu time.

Reply:
If we're satisfied that the memory usage after using certain thresholds is acceptable, I don't get why it needs to be a user facing tunable. A kill switch makes sense for ppl who want zero increase in memory usage. But I don't see how an end user could understand this config and figure out how much impact to memory usage changing it will have.
You can benchmark and tune it in future without making it a user config as well.
While we have other examples of configs in the codebase that are not end-user friendly and are tuned based on benchmark results, we should still try to avoid it where we can.

Reply:
OK, I'll remove the tunable config option (for now, we may return to that after some other PRs are merged). I don't think kill switch is reasonable though. The memory consumption increase is so small with the threshold that making any concern out of this is an unnecessary complexity. The kill switch may work if the threshold is bigger, but, as written above, we may return to that.